internal/worker/rundynamic.go

```diff
@@ -50,20 +53,27 @@ func RunDynamicAnalysis(pkg *pkgecosystem.Pkg, sbOpts []sandbox.Option) (result.
 	var lastStatus analysis.Status
 	var lastError error
 	for _, phase := range pkg.Manager().RunPhases() {
-		result, err := dynamicanalysis.Run(sb, pkg.Command(phase))
+		startTime := time.Now()
+		phaseResult, err := dynamicanalysis.Run(sb, pkg.Command(phase))
 		lastRunPhase = phase
 
+		runDuration := time.Since(startTime)
+		durationString := fmt.Sprintf("%.1fs", runDuration.Seconds())
+
 		if err != nil {
 			// Error when trying to actually run; don't record the result for this phase
 			// or attempt subsequent phases
 			lastStatus = ""
 			lastError = err
+			log.Warn("Analysis failed after "+durationString, "phase", phase)
```

Comment:
`runDuration` should be included at the end with a key (`..., "duration", runDuration`) so it is available in the structured logging.

`durationString` in the message itself is optional.

(Same goes for the other logging calls as well)

Reply:
Also these log messages may need to have ecosystem, package and version included.

Reply:
Done